File: examples/csharp/Player/Agg/PlayerRouter.cs
using Google.Protobuf;
using Angzarr;
using Angzarr.Client;
using Angzarr.Examples;
using Player.Agg.Handlers;

namespace Player.Agg;

/// <summary>
/// Functional router for Player aggregate.
/// </summary>
public static class PlayerRouter
{
    public static CommandRouter Create()
    {
        return new CommandRouter("player", eb => PlayerState.FromEventBook(eb))
            .On<RegisterPlayer>((cmd, state) => RegisterHandler.Handle(cmd, (PlayerState)state))
            .On<DepositFunds>((cmd, state) => DepositHandler.Handle(cmd, (PlayerState)state))
            .On<WithdrawFunds>((cmd, state) => WithdrawHandler.Handle(cmd, (PlayerState)state))
            .On<ReserveFunds>((cmd, state) => ReserveHandler.Handle(cmd, (PlayerState)state))
            .On<ReleaseFunds>((cmd, state) => ReleaseHandler.Handle(cmd, (PlayerState)state))
            .On<TransferFunds>((cmd, state) => TransferHandler.Handle(cmd, (PlayerState)state));
    }
}

File: examples/java/player/agg/src/main/java/dev/angzarr/examples/player/PlayerRouter.java
// DOC: This file is referenced in docs/docs/components/aggregate.mdx
//      Update documentation when making changes to router patterns.
package dev.angzarr.examples.player;

import dev.angzarr.client.CommandRouter;
import dev.angzarr.examples.player.handlers.*;
import dev.angzarr.examples.player.state.PlayerState;
import dev.angzarr.examples.*;

/**
 * Functional router for Player aggregate.
 *
 * <p>Alternative to the OO annotation-based approach in Player.java.
 * Both patterns produce identical behavior - choose based on team preference.
 */
public final class PlayerRouter {

    private PlayerRouter() {}

    // docs:start:command_router
    public static CommandRouter<PlayerState> create() {
        return new CommandRouter<PlayerState>("player", StateBuilder::fromEventBook)
            .on("RegisterPlayer", RegisterPlayer.class, RegisterHandler::handle)
            .on("DepositFunds", DepositFunds.class, DepositHandler::handle)
            .on("WithdrawFunds", WithdrawFunds.class, WithdrawHandler::handle)
            .on("ReserveFunds", ReserveFunds.class, ReserveHandler::handle)
            .on("ReleaseFunds", ReleaseFunds.class, ReleaseHandler::handle);
    }
    // docs:end:command_router
}
